evm/src/cpu/kernel/interpreter.rs

```diff
@@ -103,6 +123,12 @@ pub(crate) fn run<'a>(
     Ok(interpreter)
 }
 
+enum InterpreterMemOpKind {
+    Push(usize),                      // We need to provide the context
+    Pop(U256, usize), // If you pop a certain value, you need to push it back to the correct context when reverting
+    Write(U256, usize, usize, usize), // If we write a value at a certain address, we need to write the old value back when reverting
```

Comment:
very nit: because of the automatic space alignement, it may be better to have them as 
```suggestion
    /// We need to provide the context.
    Push(usize),
    /// If we pop a certain value, we need to push it back to the correct context when reverting.
    Pop(U256, usize),
    /// If we write a value at a certain address, we need to write the old value back when reverting.
    Write(U256, usize, usize, usize),
```
note that I've added an extra `/` and updated the second comment for consistency with the others

Reply:
Also may be nice to have a descriptive documentation at the top of the enum